File: Android/java/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/storage/StorageDialogFragment.java
package com.didichuxing.doraemonkit.weex.storage;

import android.app.Dialog;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.didichuxing.doraemonkit.weex.R;


public class StorageDialogFragment extends DialogFragment {

    public static final String KEY_STORAGE_INFO = "key_storage_info";

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.dk_item_storage_dialog, container);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (getArguments() != null) {
            StorageInfo info = (StorageInfo) getArguments().getSerializable(KEY_STORAGE_INFO);
            if (info != null) {
                TextView tv_key = view.findViewById(R.id.tv_name);
                TextView tv_value = view.findViewById(R.id.tv_value);
                tv_key.setText(info.key);
                tv_value.setText(info.value);
            }
        }
    }

    @Override
    public void onStart() {
        super.onStart();
        Dialog dialog = getDialog();
        if (dialog != null) {
            DisplayMetrics dm = new DisplayMetrics();
            getActivity().getWindowManager().getDefaultDisplay().getMetrics(dm);
            dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        }
    }

}

File: Android/kotlin/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/storage/fragment/StorageDialogFragment.kt
package com.didichuxing.doraemonkit.weex.storage.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import com.didichuxing.doraemonkit.weex.R
import com.didichuxing.doraemonkit.weex.storage.StorageInfo
import kotlinx.android.synthetic.main.dk_item_storage_dialog.*

/**
 * Transformed by alvince on 2020/7/1
 *
 * @author alvince.zy@gmail.com
 */
class StorageDialogFragment : DialogFragment() {

    companion object {
        const val KEY_STORAGE_INFO = "key_storage_info"
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
        inflater.inflate(R.layout.dk_item_storage_dialog, container)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        arguments?.getSerializable(KEY_STORAGE_INFO)
            ?.let { it as? StorageInfo }
            ?.also { info ->
                view.apply {
                    tv_name.text = info.key
                    tv_value.text = info.value
                }
            }
    }

    override fun onStart() {
        super.onStart()
        dialog?.also {
            it.window?.setLayout(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
        }
    }

}